src/libYARP_dev/include/yarp/dev/IFrameSource.h

```diff
@@ -0,0 +1,200 @@
+};
+
+class FrameEditor
+{
+    std::map<std::string, yarp::math::FrameTransform> storage;
+    yarp::math::FrameTransform nullframe;
+public:
+    const auto begin() const { return storage.begin(); }
+    const auto end() const { return storage.end(); }
+    bool insertUpdate(const yarp::math::FrameTransform& frame)
+    {
+        storage[frame.frameId] = frame;
+        return true;
+    }
+
+    robotology::Result<yarp::math::FrameTransform> get(const std::string& frameid)
+    {
+        if (storage.find(frameid) == storage.end())
+        {
+            return { false, 0, "", nullframe };
+        }
+        return { true, 0, "", storage[frameid] };
+    }
+};
+
+class YARP_dev_API yarp::dev::IFrameSource
+{
+    enum ConnectionType
+    {
+        DIRECT,
+        INVERSE,
+        UNDIRECT,
+        DISCONNECTED
+    };
+    ConnectionType getConnectionType(const std::string& target_frame, const std::string& source_frame, std::string* commonAncestor = nullptr);
+
+    robotology::Result<yarp::math::FrameTransform> getChainedTransform(const std::string& target_frame_id, const std::string& source_frame_id);
+
+    virtual bool frameExistsRaw(const std::string &frame_id);
+
+    FrameEditor frameContainer;
+
+    virtual robotology::Result<std::string> getParentRaw(const std::string& frame_id);
+
+protected:
+    std::function<bool(IFrameSource*)> callback;
+    FrameEditor* getFrameContainer();
+    virtual void updateFrameContainer(FrameEditor& frameContainer) = 0;
+    virtual bool callbackPrepare() = 0;
+    virtual bool callbackDismiss() = 0;
+
+    
+public:
+    enum errors
+    {
+        OK = 0,
+        FRAME_NOT_FOUND,
+        TARGET_NOT_FOUND,
+        SOURCE_NOT_FOUND,
+        FRAMES_NOT_CONNECTED,
+        WRONG_INPUT_FORMAT,
+        TIMEOUT,
+        CALLBACK_ACTIVE,
+        UNDEFINED
+    };
+    virtual ~IFrameSource();
+
+    bool unsetCallback();
+
+    bool setCallback(std::function<bool(IFrameSource*)> cb);
+    /**
+    Creates a debug string containing the list of all registered frames.
+    * @param all_frames the returned string containing the frames
+    * @return std::string
+    */
+    virtual std::string allFramesAsString();
+
+    /**
+    Test if a transform exists.
+    * @param target_frame_id the name of target reference frame
+    * @param source_frame_id the name of source reference frame
+    * @return true/false
+    */
+    virtual robotology::Result<bool> canTransform(const std::string &target_frame, const std::string &source_frame);
+
+    /**
+    Check if a frame exists.
+    * @param frame_id the frame to be searched
+    * @param target_frame_id the name of target reference frame
+    * @param source_frame_id the name of source reference frame
+    * @return true/false
+    */
+    virtual bool frameExists(const std::string &frame_id);
+
+    /**
+    Gets a vector containing all the registered frames.
+    * @param ids the returned vector containing all frame ids
+    * @return true/false
```

Comment:
Documentation seems to be outdated w.r.t. to the methods.

Reply:
yes, see the previous comment or the notes section of the pr